signingscript/src/signingscript/sign.py

```diff
@@ -1548,3 +1549,50 @@ async def sign_debian_pkg(context, path, fmt, *args, **kwargs):
             output_file.write(signed_file["content"])
     await _create_tarfile(context, path, all_file_names, compression, tmp_dir=tmp_dir)
     return path
+
+
+async def rcodesign_notarize(paths, notarization_workdir, creds_path):
+    NOTARIZATION_EXTENSIONS = ("app", "pkg")
+    for app in paths:
+        if app.split(".")[-1].lower() not in NOTARIZATION_EXTENSIONS:
+            raise SigningScriptError("Unable to notarize app format: %s" % app)
+        app_path = os.path.join(notarization_workdir, app)
+        command = [
+            "rcodesign",
+            "notary-submit",
+            "--staple",
```

Comment:
This causes rcodesign to block on notarization completing. This is a notable change from our current process, and will tie up available workers in whatever pool this runs in.

I think this is OK to do, _however_, this argues for creating a new pool that is exclusively responsible for notarization and stapling (to avoid other signing types getting blocked or delayed if there's a spike in notarization requests, or Apple is slow, etc, etc.). This will end up being a nice simplification and clean up of the current graph (where we have the confusing "sign1" -> "poll" -> "sign3" flow).

(I don't think we need to use the current model of separate workers for notarization and stapling, as long as we do use a separate worker type here, and make sure its `maxCapacity` is set fairly high.)

In an ideal world, each `signingscript` that is handling notarization would be able to claim multiple tasks (since most of its work is just sleeping and waiting on Apple) -- but that is well into the realm of future improvements, and absolutely not part of this initial work.

Reply:
Let's get some measurements on how quick this goes with adhoc, then we'll have a better idea of how bad it could be.